packages/x-date-pickers-pro/src/DateRangeCalendar/useDragRange.ts

```diff
@@ -169,6 +169,8 @@ const useDragRangeEvents = <TDate>({
     event.stopPropagation();
     setIsDragging(false);
     setRangeDragDay(null);
+    // make sure the focused element is the element where drop ended
+    event.currentTarget.focus();
```

Comment:
@joserodolfofreitas This should fix your mentioned issue of focus remaining on the previously selected date—where the drag started.